packages/curve-ui-kit/src/shared/ui/DataTable.tsx

```diff
@@ -0,0 +1,155 @@
+import { useRef } from 'react'
+import Typography from '@mui/material/Typography'
+import TableHead from '@mui/material/TableHead'
+import TableCell from '@mui/material/TableCell'
+import TableBody from '@mui/material/TableBody'
+import Table from '@mui/material/Table'
+import TableRow from '@mui/material/TableRow'
+import { SizesAndSpaces } from '../../themes/design/1_sizes_spaces'
+import { useIntersectionObserver } from 'ui'
+import {
+  Cell,
+  Column,
+  ColumnDef,
+  flexRender,
+  getCoreRowModel,
+  getSortedRowModel,
+  Header,
+  Row,
+  SortingState,
+  useReactTable,
+} from '@tanstack/react-table'
+import { useSortFromQueryString } from '../../hooks/useSortFromQueryString'
+import { ArrowDownIcon } from '../icons/ArrowIcon'
+
+const { Sizing, Spacing, MinWidth } = SizesAndSpaces
+
+const getAlignment = <T extends any>({ columnDef }: Column<T>) => (columnDef.meta?.type == 'numeric' ? 'right' : 'left')
+
+const DataCell = <T extends unknown>({ cell }: { cell: Cell<T, unknown> }) => {
```

Comment:
`const DataCell = ({ cell }: { cell: Cell<any, unknown> }) => {`

Reply:
As I said above this would disable typescript checks